builds/checkin/rust-test-modules.yaml

```diff
@@ -0,0 +1,72 @@
+trigger: none
+pr:
+  branches:
+    include:
+      - master
+      - release/*
+jobs:
+  ################################################################################
+  - job: check_run_pipeline
+  ################################################################################
+    displayName: Check pipeline preconditions (changes ARE in builds or mqtt)
+    pool:
+      vmImage: "ubuntu-16.04"
+    steps:
+      - checkout: self
+        submodules: false
+        fetchDepth: 3
+      - bash: |
+          git log -m -1 --name-only --first-parent --pretty="" | egrep -i '^(builds|test/modules/GenericMqttTester|test/rust-test-util)'
+          if [[ $? == 0 ]]; then
+            echo "Detected changes inside builds or rust test module folders"
+            echo "##vso[task.setvariable variable=RUN_PIPELINE;isOutput=true]TRUE"
+          fi
+        displayName: Check changes in sources
+        name: check_files
+
+  ################################################################################
+  - job: linux_amd64
+  ################################################################################
+    displayName: Linux amd64
+    dependsOn: check_run_pipeline
+    condition: eq(dependencies.check_run_pipeline.outputs['check_files.RUN_PIPELINE'], 'true')
+    pool:
+      vmImage: "ubuntu-16.04"
+    steps:
+      - checkout: self
+        submodules: false # mqtt broker does not use submodules
+        fetchDepth: 3
+      - script: echo "##vso[task.setvariable variable=RUST_BACKTRACE;]1"
+        displayName: Set env variables
+      - bash: scripts/linux/generic-rust/install.sh --project-root "test/modules/GenericMqttTester"
+        displayName: Install Rust
+      - bash: scripts/linux/generic-rust/build.sh --project-root "test/modules/GenericMqttTester" --packages "Cargo.toml" --manifest-path
+        displayName: Build Generic Mqtt Tester Module
+      - bash: mqtt/build/linux/test.sh --report test-results.xml
+        displayName: Test Generic Mqtt Tester Module
+      - task: PublishTestResults@2
+        displayName: Publish test results
+        inputs:
+          testResultsFormat: "JUnit"
+          testResultsFiles: "**/test-results.xml"
+          failTaskOnFailedTests: true
+        condition: succeededOrFailed()
+
+  ################################################################################
+  - job: style_check
+  ################################################################################
```

Comment:
nit: does it make sense to have style check separate for this small tool? I believe since we removed Windows support and now build for one platform, it might be better to combine style+build+test. (like it's done in watchdog)